docs/cloud-integration.md

```diff
@@ -0,0 +1,953 @@
+
+
+## <a name="quick_start"></a>Quick Start
+
+Provided the relevant libraries are on the classpath, and Spark is configured with your credentials,
+objects in an object store can be can be read or written through URLs which uses the name of the
+object store client as the schema and the bucket/container as the hostname.
+
+
+### Dependencies
+
+The Spark application neeeds the relevant Hadoop libraries, which can
+be done by including the `spark-cloud` module for the specific version of spark used.
+
+The Spark application should include <code>hadoop-openstack</code> dependency, which can
+be done by including the `spark-cloud` module for the specific version of spark used.
+For example, for Maven support, add the following to the <code>pom.xml</code> file:
+
+{% highlight xml %}
+<dependencyManagement>
+  ...
+  <dependency>
+    <groupId>org.apache.spark</groupId>
+    <artifactId>spark-cloud_2.11</artifactId>
+    <version>${spark.version}</version>
+  </dependency>
+  ...
+</dependencyManagement>
+{% endhighlight %}
+
+If using the Scala 2.10-compatible version of Spark, the artifact is of course `spark-cloud_2.10`.
+
+### Basic Use
+
+
+
+To refer to a path in Amazon S3, use `s3a://` as the scheme (Hadoop 2.7+) or `s3n://` on older versions.
+
+{% highlight scala %}
+sparkContext.textFile("s3a://landsat-pds/scene_list.gz").count()
+{% endhighlight %}
+
+Similarly, an RDD can be saved to an object store via `saveAsTextFile()`
+
+
+{% highlight scala %}
+val numbers = sparkContext.parallelize(1 to 1000)
+
+// save to Amazon S3 (or compatible implementation)
+numbers.saveAsTextFile("s3a://testbucket/counts")
+
+// Save to Azure Object store
+numbers.saveAsTextFile("wasb://testbucket@example.blob.core.windows.net/counts")
+
+// save to an OpenStack Swift implementation
+numbers.saveAsTextFile("swift://testbucket.openstack1/counts")
+{% endhighlight %}
+
+That's essentially it: object stores can act as a source and destination of data, using exactly
+the same APIs to load and save data as one uses to work with data in HDFS or other filesystems.
+
+Because object stores are viewed by Spark as filesystems, object stores can
+be used as the source or destination of any spark work —be it batch, SQL, DataFrame,
+Streaming or something else.
+
+The steps to do so are as follows
+
+1. Use the full URI to refer to a bucket, including the prefix for the client-side library
+to use. Example: `s3a://landsat-pds/scene_list.gz`
+1. Have the Spark context configured with the authentication details of the object store.
+In a YARN cluster, this may also be done in the `core-site.xml` file.
+1. Have the JAR containing the filesystem classes on the classpath —along with all of its dependencies.
+
+### <a name="dataframes"></a>Example: DataFrames
+
+DataFrames can be created from and saved to object stores through the `read()` and `write()` methods.
+
+{% highlight scala %}
+import org.apache.spark.SparkConf
```

Comment:
I had some examples, I remove them. I can put them back. But as you note, it's only a URL; the example is there to make it clear. I can just cut it back to "use it wherever you would any other path"